app/lib/client/connection_checker.dart

```diff
@@ -8,7 +8,9 @@ enum ConnectionStatus { connected, disconnected }
 class CustomConnectionChecker {
   ConnectionStatus _status = ConnectionStatus.disconnected;
   final _statusController = StreamController<ConnectionStatus>.broadcast();
-  final dio = Dio();
+  final dio = Dio(
+    BaseOptions(validateStatus: (status) => status != null && (status == 200 || status == 302 || status == 503))
```

Comment:
This statement is way to complicated for what it checks?

Reply:
This just defines what are the valid status codes because normally, if code 503 is returned from the server, Dio itself throws an exception. That's the reason that's there. It just modifies the default option, to not throw an Exception on 503.